cmd/auto-pause/auto-pause.go

```diff
@@ -46,18 +46,25 @@ var runtime = "docker"
 func main() {
 	// TODO: #10595 make this configurable
 	const interval = time.Minute * 1
+	// interval must > 0 or the ticker will panic
+	if interval <= 0 {
+		exit.Message(reason.Usage, "Sorry, the auto-pause interval must be greater than 0, got {{.interval}}", out.V{"interval": interval.String()})
+	}
+	idleTimeout := time.NewTicker(interval)
+	defer idleTimeout.Stop()
+
 	// channel for incoming messages
 	go func() {
 		for {
 			// On each iteration new timer is created
 			select {
-			// TODO: #10596 make it memory-leak proof
-			case <-time.After(interval):
+			case <-idleTimeout.C:
 				runPause()
 			case <-unpauseRequests:
 				fmt.Printf("Got request\n")
 				if runtimePaused {
 					runUnpause()
+					idleTimeout.Reset(interval)
```

Comment:
There is a race condition here still. We can have a tick between `runUnpause` and `idleTimeout.Reset(interval)`. In this case on next `for` iteration we'll have an event in `idleTimeout.C` and do pause.

Reply:
@ilya-zuyev Yes, I also think of that possibility, but I am not sure whether reset() could clear existing & pending tick? I do some investigation and could not find the answer. if it is indeed the case, we should recreate a ticker....

Reply:
Also, an more easier way is to reset the ticker before `runUnpause()` and then reset it again after
```
idleTimeout.Reset(/*very long inverval*/)
runUnpause()
idleTimeout.Reset(interval)
```
But this may introduce a potential problem of what is the proper value for the long interval.
We can discuss offline to be more efficient.

Reply:
If we recreate a timer, we cannot use defer to stop the timer? Since the defer target is determined at the time defer is executed? Adding stop() on each exist route may cause the code ugly either.

Another way is:
```
idleTimeout.Reset(/* longer interval*/)
runUnpause()
idleTimeout.Reset(interval)
```
But it introduces a protential problem: what is the long internal number?
We can discuss on slack offline.

Reply:
Or maybe we can try drain the ticker channel once after reset?
```
runUnpause()
idleTimeout.Reset(interval)
for {
    select {
    case <-idleTimeout.C:
    default:
       break
    }
}
```

Reply:
@hetong07 Sorry for late response. Yeah, this solution with draining `idleTimeout.C` looks good to me